src/modules/service-catalog/ServiceCatalogList.tsx

```diff
@@ -25,82 +27,75 @@ const StyledGrid = styled(Grid)`
+  const [searchInputValue, setSearchInputValue] = useState("");
+
   const { t } = useTranslation();
+  const notify = useNotify();
 
-  if (error) {
-    throw error;
+  const {
+    serviceCatalogItems,
+    meta,
+    count,
+    isLoading,
+    errorFetchingItems,
+    fetchServiceCatalogItems,
+  } = useServiceCatalogItems();
+
+  if (errorFetchingItems) {
+    notify({
+      title: t(
+        "service-catalog.service-list-error-title",
+        "Services couldn't be loaded"
+      ),
+      message: t(
+        "service-catalog.service-list-error-message",
+        "Give it a moment and try it again"
+      ),
+      type: "error",
+    });
+    throw errorFetchingItems;
   }
 
+  const debouncedUpdateServiceCatalogItems = useMemo(
+    () => debounce(fetchServiceCatalogItems, 300),
+    [fetchServiceCatalogItems]
+  );
+
   useEffect(() => {
-    async function fetchData() {
-      setIsLoading(true);
-      try {
-        const response = await fetch(
-          currentCursor
-            ? `/api/v2/help_center/service_catalog/items?page[size]=${PAGE_SIZE}&${currentCursor}`
-            : `/api/v2/help_center/service_catalog/items?page[size]=${PAGE_SIZE}`
-        );
-        const data = await response.json();
-        if (response.ok) {
-          setMeta(data.meta);
-          setServiceCatalogItems(data.service_catalog_items);
-          setCount(data.count);
-          setIsLoading(false);
-        }
-        if (!response.ok) {
-          setIsLoading(false);
-          throw new Error(`HTTP error! status: ${response.status}`);
-        }
-      } catch (error) {
-        setIsLoading(false);
-        notify({
-          title: t(
-            "service-catalog.service-list-error-title",
-            "Services couldn't be loaded"
-          ),
-          message: t(
-            "service-catalog.service-list-error-message",
-            "Give it a moment and try it again"
-          ),
-          type: "error",
-        });
-        setError(error);
-      }
-    }
-    fetchData();
-  }, [currentCursor, notify, t]);
+    fetchServiceCatalogItems("", null);
+  }, [fetchServiceCatalogItems]);
+
+  useEffect(() => {
+    return () => debouncedUpdateServiceCatalogItems.cancel();
+  }, [debouncedUpdateServiceCatalogItems]);
 
   const handleNextClick = () => {
     if (meta && meta.after_cursor) {
-      setCurrentCursor(`page[after]=${meta.after_cursor}`);
+      fetchServiceCatalogItems(
+        searchInputValue,
+        "page[after]=" + meta.after_cursor
+      );
     }
   };
 
   const handlePreviousClick = () => {
     if (meta && meta.before_cursor) {
-      setCurrentCursor(`page[before]=${meta?.before_cursor}`);
+      fetchServiceCatalogItems(
+        searchInputValue,
+        "page[before]=" + meta.before_cursor
+      );
+    }
+  };
+
+  const handleInputChange = (value: string) => {
+    setSearchInputValue(value);
+    if (value.length >= 3) {
+      debouncedUpdateServiceCatalogItems(value, null);
```

Comment:
I suggested this change in the last review, but now trying it I realize it is wrong 😄 
If the user searches for something with more than 3 characters and then removes them, we are not fetching the data anymore. This means that we continue to show old results, even when the field is empty.

The previous version was better even if not ideal, maybe we should consult with Guntis on how it should work or maybe we can remove the limits on the characters